File: src/1544-Make-The-String-Great/make_the_string_great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
  let stack = [];
        
  for (const c of s) {
    if (stack.length && Math.abs(stack[stack.length - 1].charCodeAt(0) - c.charCodeAt(0)) === 32) {
      stack.pop();
    } else {
      stack.push(c);
    }
  }
  
  let ans = stack.join('');
  return ans
};

const s = s = "leEeetcode"

console.log(makeGood(s))

File: src/1544-Make-The-String-Great/make_the_string_great.py
from typing import List

class Solution:
    def makeGood(self, s: str) -> str:
        stack: List[str] = []
        
        for c in s:
            if not stack:
                stack.append(c)
            
            elif abs(ord(stack[-1]) - ord(c)) == 32:
                stack.pop()
            
            else:
                stack.append(c)

        ans: str = ''.join(stack)

        return ans
    
if __name__ == "__main__":
    s: str = "leEeetcode"
    solution = Solution()
    
    print(solution.makeGood(s))